rest/resource.go

```diff
@@ -0,0 +1,107 @@
+package rest
+
+import (
+	"github.com/gertd/go-pluralize"
+	"github.com/gin-gonic/gin"
+	"github.com/gobeam/stringy"
+	"net/http"
+	"regexp"
+	"strings"
+)
+
+type ResourceHandlers map[string]gin.HandlerFunc
+
+type ResourceOptions struct {
+	Param   string
+	Exclude []string
+}
+
+func Resource(group *gin.RouterGroup, handlers ResourceHandlers, options ResourceOptions) *gin.RouterGroup {
+	name := getNameFromPath(group.BasePath())
+	options.Param = getRouteParam(name, options.Param)
+
+	handleResourceMethod(group, &handlers, options)
+
+	return group
+}
+
+type routeMethod struct {
+	Name       string
+	HttpMethod string
+	Param      string
+}
+
+type routeMethodList map[string]routeMethod
+
+func allowedRouteMethods(param string) routeMethodList {
+	return routeMethodList{
+		"Index":  routeMethod{Name: "Index", HttpMethod: http.MethodGet},
+		"Create": routeMethod{Name: "Create", HttpMethod: http.MethodPost},
+		"Show":   routeMethod{Name: "Show", HttpMethod: http.MethodGet, Param: param},
+		"Update": routeMethod{Name: "Update", HttpMethod: http.MethodPut, Param: param},
+		"Delete": routeMethod{Name: "Delete", HttpMethod: http.MethodDelete, Param: param},
```

Comment:
Personally, I'd go with having enum-style indexes, rather than strings, to avoid possibility of having a typo:

```go
const (
	MethodIndex = iota
	MethodCreate
	MethodShow
	MethodUpdate
	MethodDelete
)
```

But I agree your way the code looks more readable.